webapp/channels/src/components/sidebar/sidebar_category/sidebar_category.tsx

```diff
@@ -316,7 +315,8 @@ export default class SidebarCategory extends React.PureComponent<Props, State> {
 
         let displayName = category.display_name;
         if (category.type !== CategoryTypes.CUSTOM) {
-            displayName = localizeMessage(`sidebar.types.${category.type}`, category.display_name);
+            const message = categoryNames[category.type as keyof typeof categoryNames];
```

Comment:
I spent a while fighting with this, but I could not get it to stop complaining that `categoryNames` didn't have an entry for `'custom'` even though the above line says that it can't be that value specifically. It also didn't work if I changed the above check to `category.type in categoryNames` either, so I eventually just gave up and went with the type assertion